dangerzone/isolation_provider/qubes.py

```diff
@@ -0,0 +1,173 @@
+                    stdout=subprocess.PIPE,
+                    startupinfo=get_subprocess_startupinfo(),
+                )
+
+            untrusted_n_pages = p.stdout.read(2)
+            n_pages = int.from_bytes(untrusted_n_pages, byteorder="big", signed=False)
+            if n_pages == 0:
+                # FIXME: Fail loudly in that case
+                return False
+            if ocr_lang:
+                percentage_per_page = 50.0 / n_pages
+            else:
+                percentage_per_page = 100.0 / n_pages
+            for page in range(1, n_pages + 1):
+                # TODO handle too width > MAX_PAGE_WIDTH
+                # TODO handle too big height > MAX_PAGE_HEIGHT
+                untrusted_width = p.stdout.read(2)
+                untrusted_height = p.stdout.read(2)
+                width = int.from_bytes(untrusted_width, byteorder="big", signed=False)
+                height = int.from_bytes(untrusted_height, byteorder="big", signed=False)
+                untrusted_pixels = p.stdout.read(
+                    width * height * 3
+                )  # three color channels
+
+                # Wrapper code
+                with open(f"/tmp/dangerzone/page-{page}.width", "w") as f:
+                    f.write(str(width))
+                with open(f"/tmp/dangerzone/page-{page}.height", "w") as f:
+                    f.write(str(height))
+                with open(f"/tmp/dangerzone/page-{page}.rgb", "wb") as f:
+                    f.write(untrusted_pixels)
+
+                percentage += percentage_per_page
+
+                text = f"Converting page {page}/{n_pages} to pixels"
+                self.print_progress(document, False, text, percentage)
+                if stdout_callback:
+                    stdout_callback(False, text, percentage)
+
+        # TODO handle leftover code input
+        text = "Converted document to pixels"
+        self.print_progress(document, False, text, percentage)  # type: ignore [arg-type]
+        if stdout_callback:
+            stdout_callback(False, text, percentage)
+
+        # FIXME pass OCR stuff properly
+        old_environ = dict(os.environ)
+        if ocr_lang:
+            os.environ["OCR"] = "1"
+            os.environ["OCR_LANGUAGE"] = ocr_lang
+
+        asyncio.run(
+            PixelsToPDF().convert()
+        )  # TODO add progress updates on second stage
+
+        percentage = 100.0
+        text = "Safe PDF created"
+        self.print_progress(document, False, text, percentage)  # type: ignore [arg-type]
+        if stdout_callback:
+            stdout_callback(False, text, percentage)
+
+        # FIXME remove once the OCR args are no longer passed with env vars
+        os.environ.clear()
+        os.environ.update(old_environ)
+
+        shutil.move(CONVERTED_FILE_PATH, document.output_filename)
+        success = True
+
+        return success
+
+    def get_max_parallel_conversions(self) -> int:
+        return 1
+
+
+def running_on_qubes() -> bool:
```

Comment:
Replace with `dangerzone/conversion/common.py:running_on_qubes()` now that the ex-container code is now under the same python module.